internal/job/shell/shell.go

```diff
@@ -299,14 +299,17 @@ func (s *Shell) RunWithOlfactor(
 		return err
 	}
 
-	w, o := olfactor.New(s.Writer, smell)
+	w, o := olfactor.New(s.Writer, smells)
 	if err := s.executeCommand(ctx, cmd, w, executeFlags{
 		Stdout: true,
 		Stderr: true,
 		PTY:    s.PTY,
 	}); err != nil {
-		if o.Smelt() {
-			return NewOlfactoryError(smell, err)
+		// wrap err with the smells that were smelt
+		for _, smell := range smells {
+			if o.Smelt(smell) {
+				err = NewOlfactoryError(smell, err)
```

Comment:
I think the only issue here is that a linked list of OlfactoryErrors feels like the wrong data structure.

Reply:
Good point. Shall I put the trie in the error?

Reply:
Maybe? So it would be up to the caller to check for an `OlfactoryError`, then query the trie in it to find out what it smelled?

But if you do that, then I wonder... what if we used `RunWithOlfactor` to check a particular smell is _present_, rather than absent? Then it would be better if `RunWithOlfactor` returned `(*Olfactor, error)`, so the caller could query the `Olfactor` for smells directly, and the error would only be non-nil when it failed to run.

Reply:
>  Then it would be better if RunWithOlfactor returned `(*Olfactor, error)`

Yeah, after implementing this, I agree. I've done this and eliminated `OlfactoryErrors` entirely. It's more ergonomic for the caller to have more control without layers of `error` wrapping in the way.